app/controllers/events_controller.rb

```diff
@@ -139,16 +139,12 @@ def index
 
       render json: EventSerializer.new(results, options).serialized_json, status: :ok
     else
-      aggregations = params.fetch(:aggregations, "") || ""
-
-      sources = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_source(response.response.aggregations.sources.buckets) : nil
-      prefixes = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_source(response.response.aggregations.prefixes.buckets) : nil
-      citation_types = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_citation_type(response.response.aggregations.citation_types.buckets) : nil
-      relation_types = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_relation_type(response.response.aggregations.relation_types.buckets) : nil
-      registrants = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations")  ? facet_by_registrants(response.response.aggregations.registrants.buckets) : nil
-      pairings = total.positive? && aggregations.blank? || aggregations.include?("query_aggregations") ? facet_by_pairings(response.response.aggregations.pairings.buckets) : nil
```

Comment:
I think the main performance improvement on this PR is about removing `states` and `pairings` from the aggregations. And that can be easy achieved by just removing them out of the "query_aggregations" hash.

Reply:
The performance of the events API is very poor, and we will not get away with minor changes. This is just the first step.